### PR TITLE
Use CryptParams::as_ptr() in load and repair

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to #292 

@wcs1only We found a couple more places affected by this bug. This PR should resolve all of the rest of them.

### CODE REVIEW

File: src/luks2/reencrypt.rs [hunk 1]
```diff
@@ -24,17 +26,23 @@ type ReencryptProgress = unsafe extern "C" fn(size: u64, offset: u64, *mut c_voi
...
     #[allow(dead_code)]
-    luks2_params: CryptParamsLuks2Ref<'a>,
+    luks2_params: Box<CryptParamsLuks2Ref<'a>>,
     #[allow(dead_code)]
     resilience_cstring: CString,
     #[allow(dead_code)]
     hash_cstring: CString,
 }
 
+impl<'a> CryptParamsReencryptRef<'a> {
```
Comment (mulkieran) [inline src/luks2/reencrypt.rs:40]: I'm not sure why it is necessary for return type to have ```mut``` instead of ```const``` modifier.

**Review by mulkieran — COMMENTED** [review]
I'm a bit puzzled by the requirement for the mut modifier as noted inline. It kind of perplexes me a bit as to why it is necessary in the CryptParams trait methods, also.

Other than that, it looks good.

File: src/luks2/reencrypt.rs (hunk 1 above)
Comment (jbaublitz) [reply src/luks2/reencrypt.rs:40]: You're right, this doesn't have to be that way. Resolved.

**Review by mulkieran — APPROVED** [review]

**Comment by mulkieran** [comment]
rebased, should pass all tests

**Review by bmr-cymru — APPROVED** [review]

**Pull request merged**